src/braket/circuits/circuit.py

```diff
@@ -1128,6 +1130,26 @@ def to_ir(
         else:
             raise ValueError(f"Supplied ir_type {ir_type} is not supported.")
 
+    @staticmethod
+    def from_ir(source: str, inputs: Optional[Dict[str, io_type]] = None) -> Circuit:
+        """
+        Converts OpenQASM to Braket Circuit
+
+        Args:
+            source (str): OpenQASM string.
+            inputs (Optional[Dict[str, io_type]]): Inputs to the circuit.
+
+        Returns:
+            Circuit: braket sdk circuit.
+        """
+        from braket.circuits.braket_program_context import BraketProgramContext
```

Comment:
Why are we importing here?

Reply:
It would create a circular import if I import this in the beginning of the file since  BraketProgramContext uses Circuit.

Reply:
Hm I suppose that makes sense to me. I'm not sure if we have any other imports scoped like this; @speller26 what do you think?

Reply:
Ah, one of the things I hate most about Python. Don't know if there's any way around this short of throwing the entirety of `BraketProgramContext` in this file.

_That_, however, does make sense; I don't see a need to make `BraketProgramContext` public; it'll only be used internally by the `Circuit` class, so it might be better to make it a private `_BraketProgramContext` inside this file.